src/app/pages/swap/components/swap-asset-select/swap-asset-select-base.tsx

```diff
@@ -22,19 +22,21 @@ const maxAvailableTooltip =
 const sendingMaxTooltip = 'When sending max, this amount is affected by the fee you choose.';
 
 export function SwapAssetSelectBase() {
-  const { fetchToAmount, isFetchingExchangeRate, isSendingMax, onSetIsSendingMax } =
+  const { fetchQuoteAmount, isFetchingExchangeRate, isSendingMax, onSetIsSendingMax } =
     useSwapContext();
   const { setFieldValue, setFieldError, values } = useFormikContext<SwapFormValues>();
   const [amountField, amountFieldMeta, amountFieldHelpers] = useField('swapAmountBase');
   const showError = useShowFieldError('swapAmountBase');
   const [assetField] = useField('swapAssetBase');
   const navigate = useSwapNavigate();
 
-  const amountAsFiat = useAlexSdkAmountAsFiat(
-    assetField.value.balance,
-    assetField.value.price,
-    amountField.value
-  );
+  const amountAsFiat =
+    isDefined(assetField.value && amountField.value) &&
```

Comment:
This check for `isDefined` could be inside `convertInputAmountValueToFiat`

Reply:
My feedback is to the contrary 🙃  Previously `convertInputAmountValueToFiat` had optional props, which I ask be removed.

The job of this function is to `convertInputAmountValueToFiat`. It doesn't say anything about handling the fact it's `undefined`. It's `undefined` because we get data from an async layer. Why should this maths-performing function know about that? 

Handling `undefined` here is a tight coupling of those concerns, and violation of single responsibility principal.

This syntax is ugly, but the "just handle the undefined case in the function" approach, used in many places, kicks the can down the road, and is really just cheating the compiler. An approach I really want us to avoid.

Also, it makes it really easy to accidentally pass potentially undefined values (that shouldn't be) to this function, creating bugs.

Two alternative solutions to avoid this ugly syntax:
1. **Wrap the function**. Create a new function that wraps `convertInputAmountValueToFiat`. Perhaps `convertInputAmountValueToFiatWithDefaultZero` or `convertAmountToFiatIfAmountExists`, doesn't really matter. This doesn't expand the scope of the core function.
2. **Organise components differently**. Why not create a component that handles `undefined`? And returns a default if no values exist, and if they do, calls this function.

Sorry for rant, but this is an important architectural principal I believe in that is often neglected.

cc/ @alter-eggo @pete-watters